element_array_ephys/export/nwb/nwb.py

```diff
@@ -0,0 +1,629 @@
+                description=f"probe{this_probe['probe']}_shank{shank_id}",
+                location=insert_location,
+                device=device,
+            )
+
+            electrodes_query = (
+                probe.ProbeType.Electrode & this_probe & dict(shank=shank_id)
+            ).fetch(as_dict=True)
+            for electrode in electrodes_query:
+                nwbfile.add_electrode(
+                    group=electrode_group,
+                    filtering="unknown",
+                    imp=-1.0,
+                    x=np.nan,
+                    y=np.nan,
+                    z=np.nan,
+                    rel_x=electrode["x_coord"],
+                    rel_y=electrode["y_coord"],
+                    rel_z=np.nan,
+                    shank_col=electrode["shank_col"],
+                    shank_row=electrode["shank_row"],
+                    location="unknown",
+                    id_in_probe=electrode["electrode"],
+                    shank=electrode["shank"],
+                )
+
+
+def create_units_table(
+    session_key: dict,
+    nwbfile: pynwb.NWBFile,
+    units_query,
+    paramset_record,
+    name="units",
+    desc="data on spiking units"):
+    """
+
+    ephys.CuratedClustering.Unit::unit -> units.id
+    ephys.CuratedClustering.Unit::spike_times -> units["spike_times"]
+    ephys.CuratedClustering.Unit::spike_depths -> units["spike_depths"]
+    ephys.CuratedClustering.Unit::cluster_quality_label -> units["cluster_quality_label"]
+
+    ephys.WaveformSet.PeakWaveform::peak_electrode_waveform -> units["waveform_mean"]
+
+    Parameters
+    ----------
+    session_key: dict
+    nwbfile: pynwb.NWBFile
+    units_query: ephys.CuratedClustering.Unit
+    paramset_record: int
+    name: str, optional
+        default="units"
+    desc: str, optional
+        default="data on spiking units"
+    """
+
+    # electrode id mapping
+    mapping = get_electrodes_mapping(nwbfile.electrodes)
+
+    units_table = pynwb.misc.Units(name=name, description=desc)
+    for additional_attribute in ["cluster_quality_label", "spike_depths"]:
+        units_table.add_column(
+            name=units_query.heading.attributes[additional_attribute].name,
+            description=units_query.heading.attributes[additional_attribute].comment,
+            index=additional_attribute == "spike_depths",
+        )
+
+    clustering_query = (
+        ephys.EphysRecording * ephys.ClusteringTask & session_key & paramset_record
+    )
+
+    for unit in tqdm(
+        (ephys.CuratedClustering.Unit & clustering_query).fetch(as_dict=True),
+        desc=f"creating units table for paramset {paramset_record['paramset_idx']}",
+    ):
+
+        probe_id, shank_num = (
+            probe.ProbeType.Electrode
+            * ephys.ProbeInsertion
+            * ephys.CuratedClustering.Unit
+            & {"unit": unit["unit"], "insertion_number": unit["insertion_number"]}
+        ).fetch1("probe", "shank")
```

Comment:
```suggestion
        probe_id, shank_num = (
            ephys.ProbeInsertion
            * ephys.CuratedClustering.Unit
            * probe.ProbeType.Electrode
            & unit
        ).fetch1("probe", "shank")
```

Reply:
Updated the join because this order is more clear to me but the results should be the same (see [Properties of join](https://docs.datajoint.org/python/queries/07-Join.html#properties-of-join)).

Reply:
We will also need to restrict by other attributes including `subject`, `session_datetime`, and `curation_id` so I changed the restriction.  To confirm, what are the keys of the `unit` variable?